host/src/gatt.rs

```diff
@@ -138,11 +146,49 @@ pub enum GattEvent<'reference> {
     },
 }
 
-pub struct GattClient<'reference, 'resources, T: Controller, const MAX: usize, const L2CAP_MTU: usize = 27> {
-    pub(crate) services: Vec<ServiceHandle, MAX>,
+pub struct NotificationListener<'lst> {
+    pub(crate) listener: DynamicReceiver<'lst, (u16, Packet)>,
+}
+
+impl<'lst> NotificationListener<'lst> {
+    #[allow(clippy::should_implement_trait)]
+    /// Get the next (len: u16, Packet) tuple from the rx queue
+    pub fn next(&mut self) -> impl Future<Output = (u16, Packet)> + '_ {
+        self.listener.receive()
+    }
+}
+
+pub struct NotificationManager<
+    'mgr,
+    E,
+    C: Client<E>,
+    const MAX_NOTIF: usize,
+    const NOTIF_QSIZE: usize,
+    const ATT_MTU: usize,
+> {
+    pub(crate) client: &'mgr C,
+    pub(crate) rx: DynamicReceiver<'mgr, (ConnHandle, Pdu)>,
+    _e: PhantomData<E>,
+}
+
+pub struct GattClient<
+    'reference,
+    'resources,
+    T: Controller,
+    const MAX_SERVICES: usize,
+    const MAX_NOTIF: usize,
+    const NOTIF_QSIZE: usize,
+    const L2CAP_MTU: usize = 27,
+> {
+    pub(crate) known_services: RefCell<Vec<ServiceHandle, MAX_SERVICES>>,
     pub(crate) rx: DynamicReceiver<'reference, (ConnHandle, Pdu)>,
     pub(crate) ble: &'reference BleHost<'resources, T>,
     pub(crate) connection: Connection<'reference>,
+    pub(crate) request_channel: Channel<NoopRawMutex, (ConnHandle, Pdu), NOTIF_QSIZE>,
+
+    pub(crate) notification_pool: &'static PacketPool<NoopRawMutex, L2CAP_MTU, MAX_NOTIF, 1>,
```

Comment:
You should be able to use the &'static dyn GlobalPacketPool to reduce the 'generics hell' :sweat_smile:  if you want to. This should get rid of L2CAP_MTU and MAX_NOTIF parameters from GattClient.

Reply:
OK, pushed. But I haven't been able to get rid of those parameters. Otherwise, how do I build the corresponding arrays?

Reply:
Ah you're right about the notif const. But the L2CAP_MTU should be removable? I don't see it used anywhere. In any case, I think it's fine to merge.

Just rebase it on latest main and we can merge it

Reply:
> Ah you're right about the notif const. But the L2CAP_MTU should be removable? I don't see it used anywhere

it's still used in `request` to build the receive buffer. Anyway, rebased and pushed!